app/src/main/java/com/example/bottomnav/Statement.java

```diff
@@ -0,0 +1,116 @@
+package com.example.bottomnav;
+
+import org.eclipse.cdt.core.dom.ast.ExpansionOverlapsBoundaryException;
+import org.eclipse.cdt.internal.core.dom.parser.cpp.CPPASTDeclarator;
+import org.eclipse.cdt.internal.core.dom.parser.cpp.CPPASTEqualsInitializer;
+import org.eclipse.cdt.internal.core.dom.parser.cpp.CPPASTLiteralExpression;
+import org.eclipse.cdt.internal.core.dom.parser.cpp.CPPASTNamedTypeSpecifier;
+import org.eclipse.cdt.internal.core.dom.parser.cpp.CPPASTSimpleDeclSpecifier;
+import org.eclipse.cdt.internal.core.dom.parser.cpp.CPPASTSimpleDeclaration;
+
+import java.util.Optional;
+
+// Only difference between const and members: type qualifiers and values. Parsing is the same.
+public abstract class Statement {
+    protected Optional<VariableContents> variableContents;
+    protected Optional<CPPASTDeclarator> declarator;
+    protected Optional<CPPASTEqualsInitializer> initializer;
+    protected Optional<CPPASTSimpleDeclSpecifier> simpleSpecifier;
+    protected Optional<CPPASTNamedTypeSpecifier> namedTypeSpecifier;
+
+    // Initializer function must check if components of the statement are null or not before
+    // extracting the string values, else parser may break.
+    public Statement(CPPASTSimpleDeclaration declaration) {
+        CPPASTDeclarator decl = (CPPASTDeclarator) declaration.getDeclarators()[0];
+        CPPASTEqualsInitializer init = (CPPASTEqualsInitializer) decl.getInitializer();
+
+        if (decl != null) {
+            declarator = Optional.of(decl);
+        } else {
+            declarator = Optional.empty();
+        }
+        if (init != null) {
+            initializer = Optional.of(init);
+        } else {
+            initializer = Optional.empty();
+        }
+
+        /**
+         * Declaration specifiers vary!
+         * Simple Specifier: Type Specifier eg. int, float, double, etc.
+         * Named Type Specifier: Type Specifier eg. uint8_t, uint16, etc. (custom)
+         * Type Qualifier: eg. const, volatile, restrict, etc.
+         */
+
+        if (declaration.getDeclSpecifier() instanceof CPPASTSimpleDeclSpecifier) {
+            simpleSpecifier = Optional.of((CPPASTSimpleDeclSpecifier) declaration.getDeclSpecifier());
+            namedTypeSpecifier = Optional.empty();
+        } else if (declaration.getDeclSpecifier() instanceof CPPASTNamedTypeSpecifier) {
+            simpleSpecifier = Optional.empty();
+            namedTypeSpecifier = Optional.of((CPPASTNamedTypeSpecifier) declaration.getDeclSpecifier());
+        } else { // Possibility that specifier is neither simple or named type.
+            simpleSpecifier = Optional.empty();
+            namedTypeSpecifier = Optional.empty();
+        }
+    }
+
+    // Checks that both declarator and specifier exist. All variables and members must have them.
+    protected boolean isStatementValid() {
```

Comment:
I'd prefer ways to make it so that if you have an instance, it's valid - instead of getting an objecting and then needing to test for validity. We might have to think a bit about how to make this structurally work.